File: dev-docs/snippets/databases/create.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Create a new database of a given type.
export async function create(client: JsBaoClient, databaseType: string) {
  // #region example
  const db = await client.databases.create({
    title: "Product catalog",
    databaseType,
    celContext: { region: "us-east" },
  });
  // #endregion example
  return db;
}

File: dev-docs/snippets/databases/create.swift
import JsBaoClient

// Create a new database of a given type.
func create(client: JsBaoClient, databaseType: String) async throws {
  // #region example
  let db = try await client.databases.create(params: CreateDatabaseParams(
    title: "Product catalog",
    databaseType: databaseType,
    celContext: ["region": "us-east"]
  ))
  // #endregion example
  _ = db
}
